functions.go

```diff
@@ -0,0 +1,57 @@
+package memphis
+
+import (
+	"context"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+type MemphisMsg struct {
+	Headers map[string]string `json:"headers"`
+	Payload string            `json:"payload"`
+}
+
+type MemphisMsgWithError struct{
+	Headers map[string]string `json:"headers"`
+	Payload string            `json:"payload"`
+	Error string			  `json:"error"`
+}
+
+type MemphisEvent struct {
+	Messages []MemphisMsg `json:"messages"`
+}
+
+type MemphisOutput struct {
+	Messages []MemphisMsg `json:"messages"`
+	FailedMessages []MemphisMsgWithError `json:"failedMessages"`
+}
+
+type UserFunction func(string) (string, error)
+
+func CreateFunction(userFunction UserFunction){
```

Comment:
Add Go docs comments so users will be aware what this function is responsible for